src/shared/directives/conditional-show/conditional-show-form.directive.ts

```diff
@@ -60,9 +61,19 @@ export class ConditionalShowFormDirective implements OnInit, AfterViewInit, OnDe
     this.unsubscribeFromFormChanges();
   }
 
+  /*
+  * Delay the execution of evalShowHideConditions for 500ms
+  * Evaluating showHideConditions on input is inefficient as all forms are evaluated
+  * whilst the user is still typing. We are better off allowing the user to finish typing
+  * then evaluate the show hide conditions.
+  */
   private subscribeToFormChanges() {
     this.unsubscribeFromFormChanges();
-    this.formChangesSubscription = this.formGroup.valueChanges.subscribe(_ => {
+    this.formChangesSubscription = this.formGroup.valueChanges
+      .pipe(
+        debounceTime(500)
```

Comment:
Not _opposed_ to this change but I don't know if it should be part of this ticket. I think it would need architectural approval as well as it's potentially quite a big deviation in the existing behaviour.

Reply:
Agreed by @Mo-Lala84

Reply:
@Mo-Lala84 has confirmed this change is good to go. 👍